docs/development/pigs/pig-017.rst

```diff
@@ -0,0 +1,217 @@
+where it is linked, and those potential features do not replace the proposal of this PIG.
+
+Alternatives
+============
+
+The are other technical solutions that could be used to capture provenance or perform a structured logging
+in the context of Python scripting executions. The `ctapipe tools`_ come with provenance capture features
+implemented by `ctapipe.core.provenance`_, which capture the execution environment and input/output files.
+Other Python packages that provide provenance capture are `Recipy`_ and `noWorkflow`_. Both provide non-intrusive
+implementation mechanisms and provenance inspection features, but need a relational database (TinyDB, SQLite)
+to store the information and use different different non standard fixed proprietary models. In the end they
+are not highly flexible to configure the kind of information captured and the desired level of granularity.
+
+A different approach is to record provenance as structured logs, which is the one proposed in this PIG. In
+this optic we may find as the most relevant examples `Autologging`_, `Eliot`_ and `Structlog`_. Autologging
+provides an automatic basic logging for class methods calls with parameters values passed and returned, that
+is activated through class decorators. Whereas Eliot and Structlog provide Python APIs to perform a detailed
+structured logging in the form of nested dictionaries and arrays. Eliot also provides a tool to display these
+logs as a structured sequence of nested actions and subactions, as well as the parameters values involved.
+These APIs could be used to develop the custom-model provenance capture layer proposed in this PIG, but they
+add an external dependency that for our simple model we find oversized and could be easily avoided. It remains
+that if technical blockers are found during the development of this PIG, we could always explore the solutions
+offered by these APIs.
+
+Task list
+=========
+
+- General cleaning of Gammapy code-base to replace ``print()`` functions with standard Python ``logging``.
+- Define a syntax for the description provenance file and agree on a level of provenance granularity captured.
+- Define a structured format for logging as time-stamped messages in a log file and/or in output stream.
+- Prototype a non-intrusive solution to capture provenance with high-level interface tasks.
+- Additions on the configuration settings of the high-level interface to enable provenance capture.
+- Improve the prototype based on users and core developers feedback to achieve a final solution.
+- Develop a small set of tools which translate provenance logs into W3C syntax formatted files.
+- Develop a small set of tools which produce graphs of the workflow process using provenance logs.
+
+Decision
+========
+TBD
+
+
+.. _High-level interface: https://docs.gammapy.org/0.14/scripts/index.html
+.. _PIG12: https://docs.gammapy.org/0.14/development/pigs/pig-012.html
```

Comment:
PIG 12 is merged, so you can remove the URL here, and in the text use a Sphinx reference, i.e.
```
:ref:`pig-012`
```